src/native/libs/System.Globalization.Native/pal_localeStringData.c

```diff
@@ -279,26 +286,50 @@ int32_t GlobalizationNative_GetLocaleInfoString(const UChar* localeName,
             }
             break;
         case LocaleString_ThousandSeparator:
-            status = GetLocaleInfoDecimalFormatSymbol(locale, UNUM_GROUPING_SEPARATOR_SYMBOL, value, valueLength);
+            status = GetLocaleInfoDecimalFormatSymbol(locale, UNUM_GROUPING_SEPARATOR_SYMBOL, value, valueLength, NULL);
             break;
         case LocaleString_DecimalSeparator:
-            status = GetLocaleInfoDecimalFormatSymbol(locale, UNUM_DECIMAL_SEPARATOR_SYMBOL, value, valueLength);
+            status = GetLocaleInfoDecimalFormatSymbol(locale, UNUM_DECIMAL_SEPARATOR_SYMBOL, value, valueLength, NULL);
             break;
         case LocaleString_Digits:
-            status = GetDigitSymbol(locale, status, UNUM_ZERO_DIGIT_SYMBOL, 0, value, valueLength);
-            // symbols UNUM_ONE_DIGIT to UNUM_NINE_DIGIT are contiguous
-            for (int32_t symbol = UNUM_ONE_DIGIT_SYMBOL; symbol <= UNUM_NINE_DIGIT_SYMBOL; symbol++)
             {
-                int charIndex = symbol - UNUM_ONE_DIGIT_SYMBOL + 1;
-                status = GetDigitSymbol(
-                    locale, status, (UNumberFormatSymbol)symbol, charIndex, value, valueLength);
+                // Native digit can be more than one character (e.g. ccp-Cakm-BD locale which using surrogate pairs to represent the native digit).
+                // We'll separate the native digits in the returned buffer by the character '\uFFFF'.
+                int charIndex = 0;
+                int symbolLength = 0;
+
+                status = GetDigitSymbol(locale, status, UNUM_ZERO_DIGIT_SYMBOL, 0, value, valueLength, &symbolLength);
+                charIndex += symbolLength;
```

Comment:
```suggestion
                int symbolLength = 0;
                status = GetDigitSymbol(locale, status, UNUM_ZERO_DIGIT_SYMBOL, 0, value, valueLength, &symbolLength);
                
                int charIndex = symbolLength;
```

I think it's easier to read to not double initialize `charIndex`.